src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/DataAnnotationAttributesOperationFilter.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Reflection;
+
+using Swashbuckle.AspNetCore.Swagger;
+
+namespace Swashbuckle.AspNetCore.SwaggerGen
+{
+    public class DataAnnotationAttributesOperationFilter : IOperationFilter
+    {
+        public void Apply(Operation operation, OperationFilterContext context)
+        {
+            if (operation.Parameters == null)
+            {
+                return;
+            }
+
+            for (var index = 0; index < operation.Parameters.Count; index++)
+            {
+                var parameter = operation.Parameters[index];
+                if (parameter is NonBodyParameter nonBodyParam)
+                {
+                    var propertyParam = context.ApiDescription.ParameterDescriptions
+                       .Where(p => p.ModelMetadata?.ContainerType != null && p.ModelMetadata?.PropertyName != null)
+                        .FirstOrDefault(p => parameter.Name.Equals(p.Name, StringComparison.OrdinalIgnoreCase));
```

Comment:
@rthunbo, forgive me if I'm mistaken, but I believe you can't safely assume that the ApiParameterDescription from the OperationFilterContext is the same as the one from the Operation by only comparing the parameter name.

I understand it is your only hook to do so at the moment, but a Header parameter with the same name as a FromQuery parameter will result in both having (or not having, because of the order returned by FirstOrDefault) the Required property.

I'm guessing that the only way to be sure is to change the IParameter returned by [CreateParameter()](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/0e45a32d2f33c01877332f6da1516eb54a042c0a/src/Swashbuckle.AspNetCore.SwaggerGen/Generator/SwaggerGenerator.cs#L163) to contain a reference to the [paramDescription](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/0e45a32d2f33c01877332f6da1516eb54a042c0a/src/Swashbuckle.AspNetCore.SwaggerGen/Generator/SwaggerGenerator.cs#L165) it was created with. That way, you can retrieve the appropriate metadata directly from the parameter from within the operation, instead of having to lookup the parameter within the OperationFilterContext.

Reply:
@rthunbo - apologies for the late response and thanks for this contribution. You're definitely addressing a gap in the current functionality but I would tend to agree with @Erikvl87 in that it feels like core functionality as opposed to something that lives in a filter.

It's also bothering that this logic has to be essentially repeated (it or something very similar is also in SchemaExtensions.AssignValidationProperties. It all feels like a design smell (not yours but mine).

So, let me dig a little deeper before deciding on next steps. Thanks